src/Hl7.Fhir.ElementModel/TypedElementOnSourceNode.cs

```diff
@@ -248,12 +249,23 @@ private IEnumerable<TypedElementOnSourceNode> enumerateElements(Dictionary<strin
                     if (_settings.ErrorMode != TypedElementSettings.TypeErrorMode.Passthrough)
                         continue;
                 }
+                var isXhtmlNode = false;
 
-                if (instanceType == "xhtml")
+                //Special condition for ccda.
+                //If we encounter a xhtml node in a ccda document we will flatten all childnodes
+                //and use there content to build up the xml.
+                //The xml will be put in this node and children will be ignored.
+                if (instanceType == "xhtml" && info.Representation == XmlRepresentation.CdaText)
                 {
-                    var sn = SourceNode.FromNode(scan, false, new[] { typeof(IXHtml) });
-                    var ix = sn.Annotation<IXHtml>();
-                    var xml = ix?.Serialize(scan);
+                    string xml = string.Empty;
+                    foreach (var xmlChild in scan.Children())
```

Comment:
Shouldn't combining several text nodes into one be a responsibility for the underlying SourceNode?  I agree we have the type information here - so that could be a reason to only handle this here, but could this possibly be generalized so we can push it down?  Then we don't need to create another node.  This creates additional GC pressure (there are *many* such nodes in a CCDA document, and we're potentially doubling our memory footprint).

Reply:
It should be, but the SourceNode does not know it is dealing with narrative here - so it can't. Maybe the higher-level ITypedElement should indeed be implementing this, but currently, it does not work for mixed-mode narrative. See .NET API issue 1274.